### PR TITLE
use shakacode/webpacker/justin808/skip-conenthash-when-not-production-only-webpack-dev-server

### PR DESCRIPTION
**Pull request by motawfik98**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15)

### CODE REVIEW

**Review by justin808 — CHANGES_REQUESTED** [review]
> Reviewed 11 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15)* status: 11 of 16 files reviewed, 10 unresolved discussions (waiting on @motawfik98)

___
*[.gitignore, line 42 at r1](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15#-MiWsIfi5op2SVlIz8PP:-MiWsIfi5op2SVlIz8PQ:b-svr8vs) ([raw file](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/041edc172c64cf3dc3dcd362d72e4b3aab2a1f29/.gitignore#L42)):*
> ```ignore
> yarn-debug.log*
> .yarn-integrity
> .idea/
> ```

This should go in your private ~/.gitignore.

But to to see you like RubyMine!
___
*[Gemfile, line 50 at r1](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15#-MiWsRH77W5aOSBlSAYA:-MiWsRH8DM-OLXCqGEq7:bcjwyru) ([raw file](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/041edc172c64cf3dc3dcd362d72e4b3aab2a1f29/Gemfile#L50)):*
> ```Ruby
> gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
> 
> gem "webpacker", github: "shakacode/webpacker"
> ```

This is not the correct branch for the PR I gave you.
___
*[package.json, line 7 at r1](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15#-MiWsbHvBIjO2P2PROhi:-MiWsbHvBIjO2P2PROhj:b8qx9ii) ([raw file](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/3cdeeb84ea2d5786f72217b9820cf3943b147f80/package.json#L7)):*
> ```JSON
>     "@rails/webpacker": "rails/webpacker",
>     "@types/react": "^16.9.56",
>     "@types/react-dom": "^16.9.9",
> ```

Did you test removing all this?
___
*[app/packs/entrypoints/application.js, line 19 at r1](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15#-MiWsiHV2QFVGduu0bo1:-MiWsiHV2QFVGduu0bo2:bo25cm9) ([raw file](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/041edc172c64cf3dc3dcd362d72e4b3aab2a1f29/app/packs/entrypoints/application.js#L19)):*
> ```JavaScript
> // Activate Active Storage
> // import * as ActiveStorage from "@rails/activestorage"
> // ActiveStorage.start()
> ```

Whny did you add this?
___
*[app/views/layouts/application.html.erb, line 9 at r1](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15#-MiWsmn541M15cdb54O5:-MiWsmn541M15cdb54O6:b-3e8p1u) ([raw file](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/041edc172c64cf3dc3dcd362d72e4b3aab2a1f29/app/views/layouts/application.html.erb#L9)):*
> ```erb
> 
>     <%= stylesheet_link_tag    'application', media: 'all' %>
>     <%= javascript_pack_tag "application" %>
> ```

Probably not necessary, not the line above it.
___
*[bin/yarn, line 15 at r1](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15#-MiWsr_YBvZsTRNzAIMs:-MiWsr_YBvZsTRNzAIMt:b-yo3uno) ([raw file](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/041edc172c64cf3dc3dcd362d72e4b3aab2a1f29/bin/yarn#L15)):*
> ```
>   else
>     $stderr.puts "Yarn executable was not detected in the system."
>     $stderr.puts "Download Yarn at https://yarnpkg.com/en/docs/install"
> ```

probably not necessary to add this
___
*[config/webpacker.yml, line 5 at r1](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15#-MiWsuw46JUYZ-qplMdo:-MiWsuw46JUYZ-qplMdp:bg43eye) ([raw file](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/041edc172c64cf3dc3dcd362d72e4b3aab2a1f29/config/webpacker.yml#L5)):*
> ```YAML
> default: &default
>   source_path: app/packs
>   source_entry_path: entrypoints
> ```

did you test making these changes?

these changes are not necssary
___
*[config/webpacker.yml, line 28 at r1](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15#-MiWsyiiB24_eDpT1G4k:-MiWsyiiB24_eDpT1G4l:b5l3mfi) ([raw file](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/041edc172c64cf3dc3dcd362d72e4b3aab2a1f29/config/webpacker.yml#L28)):*
> ```YAML
>     port: 3035
>     # Hot Module Replacement updates modules while the application is running without a full reload
>     hmr: false
> ```

why change this?
___
*[config/webpack/base.js, line 4 at r1](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15#-MiWt1Hk3-hEOZpz2o-2:-MiWt1Hk3-hEOZpz2o-3:b-kh37fj) ([raw file](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/041edc172c64cf3dc3dcd362d72e4b3aab2a1f29/config/webpack/base.js#L4)):*
> ```JavaScript
> const customConfig = require('./webpackConfig')
> 
> module.exports = merge(webpackConfig, customConfig)
> ```

what's the point of adding this?
___
*[config/webpack/commonWebpackConfig.js, line 14 at r1](https://reviewable.io/reviews/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/15#-MiWt6As2PL1JD4dPhK2:-MiWt6As2PL1JD4dPhK3:b-759rhm) ([raw file](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/3cdeeb84ea2d5786f72217b9820cf3943b147f80/config/webpack/commonWebpackConfig.js#L14)):*
> ```JavaScript
> const commonWebpackConfig = () => (merge({}, baseClientWebpackConfig, commonOptions))
> 
> module.exports = commonWebpackConfig
> ```

you clearly didn't test anything.

**Comment by motawfik98** [comment]
HMR
Compilation with HMR
without HMR
Compilation without HMR

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
     "@rails/webpacker": "rails/webpacker",
```
Comment (justin808) [inline package.json:5]: You need rails/webpacker to be at the same version.

Comment (justin808) [reply package.json:5]: as the gem

The change made was in the JS files, so your change here to the gem only has no effect

**Comment by justin808** [comment]
see https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/pull/16

**Pull request closed without merging**